server/src/main/java/org/opensearch/action/search/SearchQueryCounters.java

```diff
@@ -49,244 +24,51 @@ final class SearchQueryCounters {
-        this.queryStringCounter = metricsRegistry.createCounter(
-            "search.query.type.querystringquery.count",
-            "Counter for the number of top level and nested queryStringQuery search queries",
-            UNIT
-        );
-        this.rangeCounter = metricsRegistry.createCounter(
-            "search.query.type.range.count",
-            "Counter for the number of top level and nested range search queries",
-            UNIT
-        );
-        this.regexpCounter = metricsRegistry.createCounter(
-            "search.query.type.regex.count",
-            "Counter for the number of top level and nested regex search queries",
-            UNIT
-        );
-        this.scriptCounter = metricsRegistry.createCounter(
-            "search.query.type.script.count",
-            "Counter for the number of top level and nested script search queries",
-            UNIT
-        );
-        this.simpleQueryStringCounter = metricsRegistry.createCounter(
-            "search.query.type.simplequerystring.count",
-            "Counter for the number of top level and nested script simple query string search queries",
-            UNIT
-        );
-        this.skippedCounter = metricsRegistry.createCounter(
-            "search.query.type.skipped.count",
-            "Counter for the number queries skipped due to error",
-            UNIT
-        );
         this.sortCounter = metricsRegistry.createCounter(
             "search.query.type.sort.count",
             "Counter for the number of top level sort search queries",
             UNIT
         );
-        this.termCounter = metricsRegistry.createCounter(
-            "search.query.type.term.count",
-            "Counter for the number of top level and nested term search queries",
-            UNIT
-        );
-        this.totalCounter = metricsRegistry.createCounter(
-            "search.query.type.total.count",
-            "Counter for the number of top level and nested search queries",
-            UNIT
-        );
-        this.wildcardCounter = metricsRegistry.createCounter(
-            "search.query.type.wildcard.count",
-            "Counter for the number of top level and nested wildcard search queries",
-            UNIT
-        );
-        this.numberOfInputFieldsCounter = metricsRegistry.createCounter(
-            "search.query.type.numberofinputfields.count",
-            "Counter for the number of input fields in the search queries",
-            UNIT
-        );
         this.queryHandlers = new HashMap<>();
-        initializeQueryHandlers();
+
     }
 
     public void incrementCounter(QueryBuilder queryBuilder, int level) {
-        Counter counter = queryHandlers.get(queryBuilder.getClass());
-        if (counter != null) {
-            counter.add(1, Tags.create().addTag(LEVEL_TAG, level));
-        } else {
-            otherQueryCounter.add(1, Tags.create().addTag(LEVEL_TAG, level));
+        String uniqueQueryCounterName = queryBuilder.getName();
```

Comment:
Should we also include queryBuilder.getClass() as part of the unique counter name, or is queryBuilder.getName() sufficient?

Reply:
Using only getName() is keeping the counter names simple and intuitive. Example: `search.query.type.bool.count` for Boolean Query Builder.